src/utils.ts

```diff
@@ -2088,67 +2161,108 @@ export class newAxis {
+      baseline = ['bottom', 'top'][this.verticalPickIdx()];
+    } else {
+      nameCoords.y -= this.offsetTitle;
+    }
+    return [nameCoords, "center", baseline, this.isVertical ? -90 : 0]
+  }
+
   private drawTicks(context: CanvasRenderingContext2D, pointHTMatrix: DOMMatrix, color: string) {
     const ticksCoords = [];
+    const ticksText: newText[] = [];
     let count = Math.max(0, this.ticks[0]);
     this.ticks.forEach((tick, idx) => {
       if (tick >= this.minValue && tick <= this.maxValue) {
+        context.setTransform(pointHTMatrix);
         let point = this.drawTickPoint(context, tick, this.isVertical, pointHTMatrix, color);
+        let text = this.labels[idx];
         ticksCoords.push(point);
-        let text = this.labels[idx]
         if (this.isDiscrete) {
           if (count == tick && this.labels[count]) { text = this.labels[count]; count++ }
           else { text = '' }
         }
-        this.drawTickText(context, text, point, pointHTMatrix);
+        ticksText.push(this.computeTickText(context, text, point, pointHTMatrix));
       }
     })
+    this.ticksFontsize = Math.min(...ticksText.map(tickText => tickText.fontsize));
+    context.resetTransform();
+    ticksText.forEach(tickText => {
+      tickText.fontsize = this.ticksFontsize;
+      tickText.width = null;
+      tickText.draw(context);
+    })
     return ticksCoords
   }
 
   private drawTickPoint(context: CanvasRenderingContext2D, tick: number, vertical: boolean, HTMatrix: DOMMatrix, color: string): newPoint2D {
     const markerOrientation = this.isVertical ? 'right' : 'up';
-    const point = new newPoint2D(tick * Number(!vertical), tick * Number(vertical), 10 / Math.abs(HTMatrix.a), 'halfLine', markerOrientation);
+    const point = new newPoint2D(tick * Number(!vertical), tick * Number(vertical), this.SIZE_END / Math.abs(HTMatrix.a), 'halfLine', markerOrientation);
     point.color = color;
     point.lineWidth /= Math.abs(HTMatrix.a);
     point.draw(context);
     return point
   }
 
-  private drawTickText(context: CanvasRenderingContext2D, text: string, point: newPoint2D, HTMatrix: DOMMatrix): void {
+  private computeTickText(context: CanvasRenderingContext2D, text: string, point: newPoint2D, HTMatrix: DOMMatrix): newText {
     const [textOrigin, textAlign, baseline] = this.tickTextPositions(point, HTMatrix);
     let textWidth = null;
-    if (textAlign == 'left') textWidth = context.canvas.width - textOrigin.x - 5;
-    if (textAlign == 'right') textWidth = textOrigin.x - 5;
-    if (textAlign == 'center') textWidth = (this.drawLength) / (this.nTicks * 1.5);
-    context.resetTransform()
-    const textParams: textParams = {width: textWidth, fontsize: this.FONT_SIZE, font: this.FONT, align: textAlign, baseline: baseline};
+    let textHeight = null;
+    if (['start', 'end'.includes(textAlign)]) {
+      textWidth = this.maxTickWidth;
+      textHeight = this.drawLength * 0.95 / this.ticks.length;
```

Comment:
Mutualization : Could set this calculus in a variable beforehand to avoid having it twice (once in if, then in else)

Reply:
Did not do that but something better.
I don't want to mutualize it because if I do, it will be computed even if it is unused.

Reply:
Are we not doing then ?

Reply:
Mutualized.
I don't understand your question

Reply:
I did it

Reply:
Funnily enough, I couldn't see it in the review tool